src/OVAL/probes/independent/system_info.c

```diff
@@ -295,6 +305,142 @@ static int get_ifs(SEXP_t *item)
         return rc;
 }
 
+#elif defined(_WIN32)
+
+#define VERSION_LEN 32
+/* Microsoft recommends to start with a 15 kB buffer for GetAdaptersAddresses */
+#define WORKING_BUFFER_SIZE 15000
+#define MAX_TRIES 3
+
+static int get_ifs(SEXP_t *item)
+{
+	/*
+	 * Use the MAKEWORD(lowbyte, highbyte) macro declared in Windef.h
+	 * to define the Windows Socket specification version we will use.
+	 * The current version of the Windows Sockets specification is version 2.2.
+	 * Version 2.2 is supported on all systems since Windows 95 OSR2.
+	 */
+	WORD wVersionRequested = MAKEWORD(2, 2);
+	LPTSTR lpMsgBuf = NULL;
+
+	WSADATA wsaData;
+	int err = WSAStartup(wVersionRequested, &wsaData);
+	if (err != 0) {
+		FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER |
+			FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+			NULL, err, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			lpMsgBuf, 0, NULL);
+		dE("Can't initialize Winsock DLL. WSAStartup failed. %s", lpMsgBuf);
+		return 1;
+	}
+
+	PIP_ADAPTER_ADDRESSES pAddresses = NULL;
+	PIP_ADAPTER_ADDRESSES pCurrAddresses = NULL;
+	DWORD dwRetVal = 0;
+	ULONG outBufLen = WORKING_BUFFER_SIZE;
+	ULONG family = AF_UNSPEC; // both IPv4 and IPv6
+	ULONG flags = GAA_FLAG_INCLUDE_PREFIX;
+	ULONG iterations = 0;
+
+	/* Try to get information about network adapters and their adresses */
+	do {
+		pAddresses = malloc(WORKING_BUFFER_SIZE);
+		dwRetVal = GetAdaptersAddresses(family, flags, NULL, pAddresses, &outBufLen);
+		if (dwRetVal == ERROR_BUFFER_OVERFLOW) {
```

Comment:
if this fails once why would it ever work again in any other try? shouldn't you increase buffer size if it fails or something like that?

Reply:
Nice catch, thatk you. I think here  should be outBufLen passed to malloc as parameter.